apps/web/src/features/spaces/components/AuthState/index.tsx

```diff
@@ -23,6 +24,10 @@ const AuthState = ({ spaceId, children }: { spaceId: string; children: ReactNode
     (member) => member.user.id === currentUser?.id && member.status === MemberStatus.DECLINED,
   )
 
+  useEffect(() => {
+    dispatch(setLastUsedSpace(spaceId))
+  }, [dispatch, spaceId])
+
```

Comment:
I added it here because we wrap this component around every space detail page so as soon as the user navigates to one of them it will be set. IMO this should be enough to cover subsequent navigations away from spaces.